File: feature-timesheet/src/main/java/fr/o80/sample/timesheet/Timesheet.java
package fr.o80.sample.timesheet;

import android.content.Context;
import android.content.Intent;

import fr.o80.sample.lib.core.Feature;
import fr.o80.sample.timesheet.presentation.ui.TimesheetActivity;

/**
 * @author Olivier Perez
 */
public class Timesheet implements Feature {
    @Override
    public void open(Context context) {
        Intent intent = TimesheetActivity.newIntent(context);
        context.startActivity(intent);
    }

    @Override
    public int getTitle() {
        return R.string.timesheet_title;
    }

    @Override
    public int getIcon() {
        return R.drawable.ic_timesheet;
    }
}

File: feature-timesheet/src/main/java/fr/o80/sample/timesheet/Timesheet.kt
package fr.o80.sample.timesheet

import android.app.Activity
import android.content.Context
import fr.o80.sample.lib.core.Feature
import fr.o80.sample.timesheet.presentation.ui.TimesheetActivity

/**
 * @author Olivier Perez
 */
class Timesheet : Feature {

    override val title: Int
        get() = R.string.timesheet_title

    override val icon: Int
        get() = R.drawable.ic_timesheet

    override fun open(context: Context) {
        val intent = TimesheetActivity.newIntent(context)
        context.startActivity(intent)
    }

    override fun notYetOpened(activity: Activity): Boolean = activity !is TimesheetActivity

}
